### PR TITLE
Expand log level extension

### PR DESCRIPTION
**Pull request by ChrisSimmons**
This PR expands on #629.  Changes:

* Adds a new extension method that also accepts an `EventId` and an `Exception`
* Update [the previous method](https://github.com/ChrisSimmons/Logging/blob/a4b901159255abbdd47cf979171838799f971f66/src/Microsoft.Extensions.Logging.Abstractions/LoggerExtensions.cs#L350) to use this new method
* Update all the level-specific methods (e.g. [`LogDebug()`](https://github.com/ChrisSimmons/Logging/blob/a4b901159255abbdd47cf979171838799f971f66/src/Microsoft.Extensions.Logging.Abstractions/LoggerExtensions.cs#L42) and [`LogWarning()`](https://github.com/ChrisSimmons/Logging/blob/a4b901159255abbdd47cf979171838799f971f66/src/Microsoft.Extensions.Logging.Abstractions/LoggerExtensions.cs#L191)) to use this new method
* Use the new method to [handle the null `logger` argument exception](https://github.com/ChrisSimmons/Logging/blob/a4b901159255abbdd47cf979171838799f971f66/src/Microsoft.Extensions.Logging.Abstractions/LoggerExtensions.cs#L366)
    * If this is not desirable, e.g. for stacktrace reasons, I can revert and resubmit the PR

### CODE REVIEW

**Comment by Eilon** [comment]
@muratg - do we want to have someone take a look at this?

**Comment by muratg** [comment]
@ChrisSimmons thanks for the PR!

@pakrym could you review when you have some time?

File: test/Microsoft.Extensions.Logging.Test/LoggerExtensionsTest.cs [hunk 1]
```diff
@@ -38,10 +38,15 @@ public void MessageOnly_LogsCorrectValues()
             logger.LogError(_state);
             logger.LogCritical(_state);
             logger.LogDebug(_state);
+            logger.Log(LogLevel.Trace, _state);
```
Comment (pakrym) [inline test/Microsoft.Extensions.Logging.Test/LoggerExtensionsTest.cs]: Add a separate test method that calls new overloads (with exception and non zero event id)

**Comment by pakrym** [comment]
Please add tests for new overloads, otherwise LGTM.

File: test/Microsoft.Extensions.Logging.Test/LoggerExtensionsTest.cs (hunk 1 above)
Comment (ChrisSimmons) [reply test/Microsoft.Extensions.Logging.Test/LoggerExtensionsTest.cs]: Sorry for the delay.  I'll get to this now.

**Comment by ChrisSimmons** [comment]
I rebased and squashed commits, which blew away your previous comments, @pakrym:

> Add a separate test method that calls new overloads (with exception and non zero event id)

I've added separate tests for the new overloads, one for each of the 10 "flavors" of the equivalent `logger.LogLEVELNAME()` extensions.  Each uses a `LogLevel_` prefix to the test method name ([Example](https://github.com/aspnet/Logging/pull/790/files#diff-099caadefe774232f5d421983c3ee853R627)).  Let me know if that naming works.

File: test/Microsoft.Extensions.Logging.Test/LoggerExtensionsTest.cs (hunk 1 above)
Comment (ChrisSimmons) [reply test/Microsoft.Extensions.Logging.Test/LoggerExtensionsTest.cs]: I've added separate tests for the new overloads, one for each of the 10 "flavors" of the equivalent `logger.LogLEVELNAME()` extensions.  Each uses a `LogLevel_` prefix to the test method name ([Example](https://github.com/aspnet/Logging/pull/790/files#diff-099caadefe774232f5d421983c3ee853R627)).  Let me know if that naming works.

(FYI, I rebased and squashed commits, @pakrym.)

**Comment by ChrisSimmons** [comment]
Also, is there any value to using a `[Theory]` to simplify these tests?  See [this](https://github.com/ChrisSimmons/Logging/commit/a113909bda2c685938e0c80a4469add8d364afcb?diff=split), for example.

**Comment by pakrym** [comment]
Totally, bring the Theories in.

File: test/Microsoft.Extensions.Logging.Test/LoggerExtensionsTest.cs [hunk 2]
```diff
@@ -628,6 +623,267 @@ public void LogValuesEventIdAndError_LogsCorrectValues()
...
+            var testLogValues = new TestLogValues()
+            {
+                Value = 1
+            };
+
+            // Act
+            logger.Log(logLevel, eventId, _exception, testLogValues.ToString());
+
+            // Assert
+            Assert.True(sink.Writes.TryTake(out var write));
```
Comment (pakrym) [inline test/Microsoft.Extensions.Logging.Test/LoggerExtensionsTest.cs]: Assert.Single(sink.Writes);
